pkg/api/validation/validation.go

```diff
@@ -2589,6 +2610,10 @@ func ValidatePodUpdate(newPod, oldPod *api.Pod) field.ErrorList {
 	mungedPod.Spec.Tolerations = oldPod.Spec.Tolerations
 	allErrs = append(allErrs, validateOnlyAddedTolerations(newPod.Spec.Tolerations, oldPod.Spec.Tolerations, specPath.Child("tolerations"))...)
 
+	// munge spec.Priority*
+	mungedPod.Spec.PriorityClassName = oldPod.Spec.PriorityClassName
```

Comment:
You already asserted that class is the same and name is nil - you don't need these?